kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -123,9 +123,13 @@ grCollectionsPanel.controller('GrNodeCtrl',
     ctrl.getCollectionQuery = path => getCollection(path);
 
     $scope.$watch('ctrl.showChildren', onValChange(show => {
-            collectionsTreeState.setState(pathId, show);
+      collectionsTreeState.setState(pathId, show);
     }));
 
+    $scope.$watch('ctrl.node.data.children', children => {
+      ctrl.children = children.filter(node => !!node.data.data);
```

Comment:
I think this filter should be a function.

Reply:
Fair point, any ideas where it should live? On the controller or in some util file somewhere?

Reply:
Yeah, just in the controller to make it obvious what it's doing.